ihmc-footstep-planning/src/main/java/us/ihmc/footstepPlanning/graphSearch/stepExpansion/ReferenceBasedIdealStepCalculator.java

```diff
@@ -57,20 +60,23 @@ public DiscreteFootstep computeIdealStep(DiscreteFootstep stanceNode, DiscreteFo
          throw new RuntimeException("Invalid side on reference plan");
       }
 
-      FramePose3D referenceFootstepPose = referenceFootstep.getFootstepPose();
-      return new DiscreteFootstep(referenceFootstepPose.getX(), referenceFootstepPose.getY(), referenceFootstepPose.getYaw(), referenceFootstep.getRobotSide());
+      double nominalIdealStepX = nominalIdealStep.getX();
+      double nominalIdealStepY = nominalIdealStep.getY();
+      double nominalIdealStepYaw = nominalIdealStep.getYaw();
+      double interpolatedYaw = AngleTools.interpolateAngle(nominalIdealStepYaw, referenceFootstep.getFootstepPose().getYaw(), referenceAlpha);
+
+      Point2D interpolatedTranslation = new Point2D(nominalIdealStepX, nominalIdealStepY);
+      Point2D referenceStepTranslation = new Point2D(referenceFootstep.getFootstepPose().getX(), referenceFootstep.getFootstepPose().getY());
```

Comment:
I think this could be 

`Point2D referenceStepTranslation = new Point2D(referenceFootstep.getFootstepPose().getPosition())`

Reply:
Actually, strike that. I would do this:

```
Pose2D nominalStepPose = new Pose2D(nominalIdealStep.getX(), nominalIdealStep.getY(), nominalIdealStep.getYaw());
Pose2D referenceStepPose = new Pose2D(referenceStep.getFootstepPose());
Pose2D interpolatedPose = new Pose2D(nominalStepPose);
interpolatedPose.interpolate(referenceStepPose, referenceAlpha);
```

Reply:
Ok sweet! Ya the pose is much better then the point, good call